### PR TITLE
Add test module for PeerDiscoveryPrimitiveImpl

### PR DESCRIPTION
**Pull request by lemonholic**
- remove old module (propagation_network)
- add unit test module for PeerDiscoveryPrimitiveImpl (unit tests are not written yet)

### CODE REVIEW

File: network/src/lib.rs [hunk 1]
```diff
@@ -29,10 +27,19 @@ pub struct Peer {
...
+    Exact(u16),
+    Any,
+}
+
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct NetworkConfig {
     /// The unique id for distinguishing the network.
     pub network_id: String,
+    /// The port used during a network operation.
+    pub port: Port,
```
Comment (junha1) [inline network/src/lib.rs:42]: In my experience, just `Option<u16>` is more common.

File: network/src/peer_discovery/primitive.rs [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+impl PeerDiscoveryPrimitive for PeerDiscoveryPrimitiveImpl {
+    async fn serve(
+        _network_config: &NetworkConfig,
+        _initially_known_peers: Vec<Peer>,
+    ) -> Result<(SharedKnownPeers, JoinHandle<Result<(), Error>>), Error> {
+        unimplemented!();
+    }
+}
+
+#[cfg(test)]
```
Comment (junha1) [inline network/src/peer_discovery/primitive.rs:17]: I think we will have bunch of tests soon. Let's move this to a separate file.

File: network/src/lib.rs (hunk 1 above)
Comment (lemonholic) [reply network/src/lib.rs]: fixed

File: network/src/peer_discovery/primitive.rs (hunk 2 above)
Comment (lemonholic) [reply network/src/peer_discovery/primitive.rs]: done

**Review by junha1 — APPROVED** [review]

**Pull request merged**